gen/groundobjectsgen.py

```diff
@@ -589,20 +589,43 @@ def __init__(
         self.tacan_registry = tacan_registry
 
     def generate(self) -> None:
+
+        # Note : Helipad are generated as neutral object in order not to interfer with capture triggers
+        neutral_country = self.m.country(self.game.neutral_country.name)
         country = self.m.country(self.game.coalition_for(self.cp.captured).country_name)
         for i, helipad in enumerate(self.cp.helipads):
             name = self.cp.name + "_helipad_" + str(i)
             logging.info("Generating helipad : " + name)
-            pad = SingleHeliPad(name=(name + "_unit"))
+            pad = InvisibleFARP(name=name)
             pad.position = Point(helipad.x, helipad.y)
             pad.heading = helipad.heading.degrees
-            # pad.heliport_frequency = self.radio_registry.alloc_uhf() TODO : alloc radio & callsign
             sg = unitgroup.StaticGroup(self.m.next_group_id(), name)
             sg.add_unit(pad)
             sp = StaticPoint()
             sp.position = pad.position
             sg.add_point(sp)
-            country.add_static_group(sg)
+            neutral_country.add_static_group(sg)
+
+            helipad.static_unit = sg
+            helipad.occupied = False
```

Comment:
Hmm. This is a bit odd. It would be better to not have mission generation alter `Game` at all.

I think instead, add a `helipads: dict[ControlPoint, list[StaticGroup]] = defaultdict(list)` member to this class (or to `Operation`) and have this populate that instead. You can do away with the `occupied` property because everything in the list is by definition unoccupied, and don't need to add mission-specific data to `Game` (`ControlPoint.helipads` reverts to being a `list[PointWithHeading]` like you had before since we don't need the additional data there).

Then, pass the `HelipadGenerator.helipads` dict to `AircraftConflictGenerator` and replace `get_free_helipad` with:

```
try:
    helipad = self.helipads[cp].pop()
except IndexError as ex:
    raise RuntimeError(f"Not enough helipads available at {cp}") from ex
```

(could be worth making a `MissionHelipads` class to contain the cruft rather than the raw dict manipulation)

That should fix the layering issue while preserving behavior. Keeps the mission generation details from affecting save compat too.